### PR TITLE
Github hook: handle creation of new tag and new branch

### PR DESCRIPTION
**Pull request by tardyp**
When a new tag or new branch is created, a push event with no commit is sent.
We must build the head_commit in that case, and then create an event with that.

## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

File: master/docs/manual/cfg-wwwhooks.rst [hunk 1]
```diff
@@ -166,10 +166,24 @@ The parameters are:
             )
 
 :guilabel:`Which events would you like to trigger this webhook?`
-    Leave the default -- ``Just the push [tag]  events`` -- other kind of events are not currently supported.
+    Click -- ``Let me select individual events``, then select ``Push`` and ``Pull request`` -- other kind of events are not currently supported.
 
 And then press the ``Add Webhook`` button.
 
+
+Github hook create 3 kinds of changes, distinguishable by their ``category`` field:
```
Comment (rjarry) [inline master/docs/manual/cfg-wwwhooks.rst]: The GitHub hook creates

File: master/docs/manual/cfg-wwwhooks.rst [hunk 2]
```diff
@@ -166,10 +166,24 @@ The parameters are:
...
+
+- ``None``: This change is a push to a branch.
+    Use ``util.ChangeFilter(category=None, repository="http://github.com/<org>/<project>")``
+- ``'tag'``: This change is a push to a tag.
+    Use ``util.ChangeFilter(category='tag', repository="http://github.com/<org>/<project>")``
+- ``'pull'``: This change is from a pull-request creation or update.
+    Use ``util.ChangeFilter(category='pull', repository="http://github.com/<org>/<project>")``
+   In this case, the :bb:step:`GitHub` step must be used instead of the standard :bb:step:`Git` in order to be able to pull GitHub's magic refs.
+   With this method, the :bb:step:`GitHub` step will always checkout the branch merged with latest master.
+   This allows to test the result of the merge instead of the just the source branch.
```
Comment (rjarry) [inline master/docs/manual/cfg-wwwhooks.rst]: Instead of **the** just the source

File: master/docs/manual/cfg-wwwhooks.rst [hunk 3]
```diff
@@ -166,10 +166,24 @@ The parameters are:
...
+
+Github hook create 3 kinds of changes, distinguishable by their ``category`` field:
+
+- ``None``: This change is a push to a branch.
+    Use ``util.ChangeFilter(category=None, repository="http://github.com/<org>/<project>")``
+- ``'tag'``: This change is a push to a tag.
+    Use ``util.ChangeFilter(category='tag', repository="http://github.com/<org>/<project>")``
+- ``'pull'``: This change is from a pull-request creation or update.
+    Use ``util.ChangeFilter(category='pull', repository="http://github.com/<org>/<project>")``
+   In this case, the :bb:step:`GitHub` step must be used instead of the standard :bb:step:`Git` in order to be able to pull GitHub's magic refs.
```
Comment (rjarry) [inline master/docs/manual/cfg-wwwhooks.rst]: Indentation ?

**Comment by tardyp** [comment]
@rjarry fixed

**Pull request merged**